source/run-agg-pipelines.txt

```diff
@@ -0,0 +1,291 @@
+   :depth: 1
+   :class: singlecol
+
+.. include:: /includes/admonitions/fact-mdb-shell-beta.rst
+
+You can run :manual:`aggregation pipelines </aggregation>` on your
+collections using the |mdb-shell|. Aggregation pipelines transform your
+documents into aggregated results based on selected 
+:manual:`pipeline stages </reference/operator/aggregation-pipeline/>`. 
+
+Common uses for aggregation include:
+
+- Grouping data by a given expression.
+
+- Calculating results based on multiple fields and storing those results
+  in a new field.
+
+- Filtering data to return a subset that matches a given criteria.
+
+- Sorting data.
+
+When you run an aggregation, |mdb-shell| conveniently outputs the
+results directly to the terminal.
+
+About the Examples
+------------------
+
+.. include:: /includes/fact-sample-data-examples.rst
+
+The examples below use the ``movies`` collection in the |service|
+:atlas:`sample_mflix </sample-data/sample-mflix/>` sample dataset. 
+
+Each document in the ``movies`` collection describes a movie:
+
+.. code-block:: javascript
+   :linenos:
+   :copyable: false
+
+   {
+     _id: 573a1390f29313caabcd4135,
+     plot: 'Three men hammer on an anvil and pass a bottle of beer around.',
+     genres: [ 'Short' ],
+     runtime: 1,
+     cast: [ 'Charles Kayser', 'John Ott' ],
+     num_mflix_comments: 1,
+     title: 'Blacksmith Scene',
+     fullplot: 'A stationary camera looks at a large anvil with a blacksmith behind it and one on either side. The smith in the middle draws a heated metal rod from the fire, places it on the anvil, and all three begin a rhythmic hammering. After several blows, the metal goes back in the fire. One smith pulls out a bottle of beer, and they each take a swig. Then, out comes the glowing metal and the hammering resumes.',
+     countries: [ 'USA' ],
+     released: 1893-05-09T00:00:00.000Z,
+     directors: [ 'William K.L. Dickson' ],
+     rated: 'UNRATED',
+     awards: { wins: 1, nominations: 0, text: '1 win.' },
+     lastupdated: '2015-08-26 00:03:50.133000000',
+     year: 1893,
+     imdb: { rating: 6.2, votes: 1189, id: 5 },
+     type: 'movie',
+     tomatoes: {
+       viewer: { rating: 3, numReviews: 184, meter: 32 },
+       lastUpdated: 2015-06-28T18:34:09.000Z
+     }
+   }
+
+The examples assume that you have already used the |mdb-shell| to switch
+to the database that contains the documents that you want to aggregate:
+
+.. code-block:: javascript
+
+   use sample_mflix
+
+Create and Run an Aggregation Pipeline
+--------------------------------------
+
+To create an aggregation pipeline, use the following syntax in the
+|mdb-shell|:
+
+.. code-block:: javascript
+   :linenos:
+
+   db.<collection>.aggregate([
+     {
+       <$stage1>
+     },
+     {
+       <$stage2>
+     }
+     ...
+   ])
+
+Example
+~~~~~~~
+
+Consider the following pipeline:
+
+.. code-block:: javascript
+   :linenos:
+
+   db.movies.aggregate([
+     {$project: {_id: 0, genres: 1, imdb: 1, title: 1} }, 
```

Comment:
[nit] Since this is an introductory guide, it may help to comment "First Stage" "Second Stage" etc as done in this example: [$group by Item Having](https://docs.mongodb.com/manual/reference/operator/aggregation/group/#group-by-item-having)